### PR TITLE
Fix ThisOldHouseIE

### PR DESCRIPTION
**Pull request by Ashish0804**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [ ] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [ ] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Closes https://github.com/yt-dlp/yt-dlp/issues/2341

### CODE REVIEW

File: yt_dlp/extractor/thisoldhouse.py [hunk 1]
```diff
@@ -41,7 +46,12 @@ class ThisOldHouseIE(InfoExtractor):
...
-        video_id = self._search_regex(
-            r'<iframe[^>]+src=[\'"](?:https?:)?//(?:www\.)?thisoldhouse\.(?:chorus\.build|com)/videos/zype/([0-9a-f]{24})',
-            webpage, 'video id')
+        if 'To Unlock This content' in webpage:
+            self.raise_login_required(method='cookies')
+        video_url = self._search_regex(
+            r'<iframe[^>]+src=[\'"]((?:https?:)?//(?:www\.)?thisoldhouse\.(?:chorus\.build|com)/videos/zype/([0-9a-f]{24})[^\'"]*)[\'"]',
+            webpage, 'video url')
+        if 'subscription_required=true' in video_url:
+            return self.url_result(self._request_webpage(video_url, display_id).url, 'Zype', display_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/thisoldhouse.py:55]:
```suggestion
            return self.url_result(self._request_webpage(HEADRequest(video_url), display_id).geturl(), 'Zype', display_id)
```

or 
```suggestion
            return self.url_result(video_url, 'Generic', display_id)
```
should work too

Comment (Ashish0804) [reply yt_dlp/extractor/thisoldhouse.py:55]: i don't want to pass to generic bcoz it's bloated.
What are the advantages of the first one?

Comment (pukkandan) [reply yt_dlp/extractor/thisoldhouse.py:55]: `.geturl()` is the right way to get redirect URL

HeadRequest makes it so the actual webpage isn't downloaded, only the headers. Check that the website doesnt block HEAD requests though

> i don't want to pass to generic bcoz it's bloated.

redirect is the first thing GenericIE tries
```py
        self.to_screen('%s: Requesting header' % video_id)

        head_req = HEADRequest(url)
        head_response = self._request_webpage(
            head_req, video_id,
            note=False, errnote='Could not send HEAD request to %s' % url,
            fatal=False)

        if head_response is not False:
            # Check for redirect
            new_url = head_response.geturl()
            if url != new_url:
                self.report_following_redirect(new_url)
                if force_videoid:
                    new_url = smuggle_url(
                        new_url, {'force_videoid': force_videoid})
                return self.url_result(new_url)
```

but your wish. Either method is fine

Comment (Ashish0804) [reply yt_dlp/extractor/thisoldhouse.py]: I should have been more verbose...
What i meant with `bcoz it's bloated` is that passing to generic prints extra stuff the cli which makes the logs bloated.
I applied the first suggestion and site indeed accepts Head request so thanks :)

File: yt_dlp/extractor/thisoldhouse.py [hunk 2]
```diff
@@ -15,6 +16,11 @@ class ThisOldHouseIE(InfoExtractor):
             'description': 'In the workshop, Tom Silva and Kevin O\'Connor build a storage bench for an entryway.',
             'timestamp': 1442548800,
             'upload_date': '20150918',
+            'duration': 674,
+            'view_count': int,
+            'average_rating': 0,
```
Comment (pukkandan) [inline yt_dlp/extractor/thisoldhouse.py:21]: this is actually 0 and not lack of data?

File: yt_dlp/extractor/thisoldhouse.py [hunk 3]
```diff
@@ -15,6 +16,11 @@ class ThisOldHouseIE(InfoExtractor):
...
+            'thumbnail': 'md5:f76278a7a7a63973a98511dc0fb0cf54',
```
Comment (pukkandan) [inline yt_dlp/extractor/thisoldhouse.py:22]: probly better to use regex here, not md5

**Review by pukkandan — APPROVED** [review]
lgtm

File: yt_dlp/extractor/thisoldhouse.py (hunk 3 above)
Comment (Ashish0804) [reply yt_dlp/extractor/thisoldhouse.py:22]: this is what running the test spit out tbh.
I just copy pasted it so test could pass

**Comment by scarng63** [comment]
Excellent working as expected.

File: yt_dlp/extractor/thisoldhouse.py (hunk 3 above)
Comment (Ashish0804) [reply yt_dlp/extractor/thisoldhouse.py]: Fixed

File: yt_dlp/extractor/thisoldhouse.py (hunk 2 above)
Comment (Ashish0804) [reply yt_dlp/extractor/thisoldhouse.py:21]: It is actually 0.
Maybe bcoz the video is old ig.

**Pull request merged**